csrc/host_ir/jit.cpp

```diff
@@ -235,130 +263,307 @@ void packOutputs(
+       int64_t_type,
+       int64_ptr_type,
+       int64_t_type,
+       int32_t_type,
+       int64_t_type,
+       tensor_ptr_type},
+      false);
+  llvm::Function::Create(
+      empty_strided_cuda_type,
+      llvm::Function::ExternalLinkage,
+      kAtEmptyStridedCudaWrapper,
+      module);
+
+  // delete_tensor function: void delete_tensor(at::Tensor* tensor)
+  auto* delete_tensor_type =
+      llvm::FunctionType::get(void_type, {tensor_ptr_type}, false);
+  llvm::Function::Create(
+      delete_tensor_type,
+      llvm::Function::ExternalLinkage,
+      kDeleteTensorFuncName,
+      module);
 
   // main function: void main(void** input_tensors, void** output_tensors)
-  llvm::FunctionType* main_type = llvm::FunctionType::get(
-      llvm::Type::getVoidTy(ctx),
-      {void_array_ptr_type, void_array_ptr_type},
-      false);
+  auto* main_type = llvm::FunctionType::get(
+      void_type, {void_array_ptr_type, void_array_ptr_type}, false);
   llvm::Function::Create(
-      main_type, llvm::Function::ExternalLinkage, kMainFuncName, mod);
+      main_type, llvm::Function::ExternalLinkage, kMainFuncName, module);
 }
 
-void compile(HostIrJitImpl* pimpl) {
+class HostIrCompileDispatcher : public OptInDispatch {
+ public:
+  HostIrCompileDispatcher(
+      llvm::IRBuilder<>& builder,
+      std::unordered_map<Val*, llvm::Value*>& val_to_value)
+      : builder_(builder), val_to_value_(val_to_value) {}
+  using OptInDispatch::handle;
+
+  // NOTE: this is just a simple example of allocate a output tensor and set it
+  // to input tensor. The whole concept is to demonstrate llvm jit works, we
+  // will change this in the future LoadStoreOp Function LLVM IR Generation
+  void handle(LoadStoreOp* load_store_op) final {
+    NVF_ERROR(
+        load_store_op->opType() == LoadStoreOpType::Set ||
+        load_store_op->opType() == LoadStoreOpType::SegmenterSet);
+    NVF_ERROR(
+        load_store_op->out()->isA<TensorView>(), "out must be a TensorView");
+    auto* in_tv = load_store_op->in()->as<Val>();
+    auto* out_tv = load_store_op->out()->as<Val>();
+    auto it = val_to_value_.find(in_tv);
+    NVF_ERROR(
+        it != val_to_value_.end(), "input tensor is not found in val_to_value");
+    llvm::Module* module = builder_.GetInsertBlock()->getParent()->getParent();
+    llvm::Value* in_tensor = it->second;
+    // allocate a new tensor
+    llvm::Function* new_tensor_func = module->getFunction(kNewTensorFuncName);
+    llvm::Value* out_tensor =
+        builder_.CreateCall(new_tensor_func, {}, "out_tensor");
+
+    // set the output tensor to the input tensor
+    llvm::Function* set_tensor_func = module->getFunction(kSetTensorFuncName);
+    builder_.CreateCall(set_tensor_func, {out_tensor, in_tensor});
+
+    // bind the output tensor to val_to_value
+    val_to_value_[out_tv] = out_tensor;
+
+    if (isDebugDumpEnabled(DebugDumpOption::HostIrJit)) {
+      printLlvmIr(builder_.GetInsertBlock()->getParent(), "LoadStoreOp");
```

Comment:
Instead of calling it in each `handle`, you can call it once in the dispatch loop: https://github.com/NVIDIA/Fuser/pull/4778/files#diff-c0568f66b12e6643f5dc2d89e0c64cfe6e3585f8c7ad3f99b95ef8b7fadddabcR520